### ISSUE TITLE
Enhancement: Option to customize the "Result" suffix appended when generating type names

### ISSUE THREAD

**Issue by vanillabrian** [ErikEJ/EFCorePowerTools#2677]
This enhancement would allow slight customization of the default return type name. We have a large amount of code written with an older version of EF and the return type names were generated slightly differently. To ease the transition to EF Core it would be nice if the type names could be generated to match the ones previously generated to minimize code changes.

Currently, the return type names generated for store procedures and functions are generated by appending the suffix "Result" to the name of the procedure or function name. The return type for the procedure **sp_ProcessSomeData** would be **sp_ProcessSomeDataResult**.

The proposal is to add an option, **Type Name Suffix**, to the Advanced Options dialog windows to specify a customized suffix instead of the default 'Result". If nothing is specified 'Result' would be used as it is in the current version. So, if "_Result" was specified as the Type Name Suffix the return type name for the procedure mentioned above would be **sp_ProcessSomeData_Result**. 

![image](https://github.com/user-attachments/assets/d19ba64d-beed-4665-9d95-cbe4c700de8c)

I have already implemented a proof of concept. Would you be open to a pull request?

**Comment by ErikEJ** [comment]
I think it is a good suggestion - but I think this could just be something you can manually add to the config file. Remember to also add to the CLI (or I can do that after your PR if you prefer)

**Comment by ErikEJ** [comment]
Cool. Testing with the CLI is very simple.

**Comment by vanillabrian** [comment]
By the way, for a PR should I include updated zip files (efptxx.exe.zip & efrevengxx.exe.zip) or just the source code changes?

**Comment by ErikEJ** [comment]
@vanillabrian Is this still something you plan to work on?

**Comment by vanillabrian** [comment]
Yes, I still plan to work on this. I was delayed by some other priorities, but I am hoping to submit a PR as soon as possible.

**Comment by ErikEJ** [comment]
@vanillabrian Going to close this for now, feel free to re-open when you find time.
